### DISCUSSION TITLE
How to make switch.altherma visible for Home Assistant under Automations & Scenes ?

### DISCUSSION THREAD

**Discussion by ap20132022** [raomin/ESPAltherma#268] [Q&A]
Hello everyone
I use the actual version to monitor and control my heat pump. Everything works fine (incl. SG control).
Last time I use a Shelly 1 to turn on and off my heat pump (the same function as switch.altherma) because I didn't have a relay for ESP. I use Shelly under automations & scenes to automatically turn HP on and off. It works very well. 
Yesterday I got some relays and tried to use them with ESP. I can add switch.altherma to my lovelace, to my picture element to control it manually. 

But I can't see switch.altherma under automations & scenes. The only thing I can use here is the Smart Grid option.
![image](https://user-images.githubusercontent.com/122316058/229692325-a2f86635-3764-4af4-8d4d-a64908aff1ed.png)

How can I make it visible to use it under automations & scences? 

Many thanks for your help.

**Comment by Alexander-Wehner** [comment]
hello, I also have a relay clamped to an ESP32, but I can't upload the software, what do I need to set?

![image](https://user-images.githubusercontent.com/112548044/230574389-a97f10de-b815-4d48-8789-ecd864752260.png)

![image](https://user-images.githubusercontent.com/112548044/230574438-a44b4b29-f314-4d94-9d8e-7626241cc3a3.png)

![image](https://user-images.githubusercontent.com/112548044/230574506-ca36a679-5202-46d7-b862-494f3095df2d.png)

greet Alex

**Comment by ap20132022** [comment]
pls download the actual code. Your code doesn't have define the SG2 (maybe you deleted it?)
My code looks different. 
![Bildschirmfoto vom 2023-04-07 11-24-41](https://user-images.githubusercontent.com/122316058/230584309-164474cd-b318-407a-8109-df41f49fa713.png)

And it's better if you open a new question. 
Here is just about SG with Home Assistant. Thanks.
